frontend/app/blueprints/public/static/js/poll_status.js

```diff
@@ -0,0 +1,62 @@
+// functions for polling job status
+async function poll(fn, fnCondition, resultParser, ms) {
+    // utility function for polling data
+    let result = await fn()
+    while (fnCondition(result)) {
+        await wait(ms)
+        result = await fn()
+    }
+    return resultParser(result)
+}
+
+function wait(ms = 2000) {
+    // wait between fetch jobs
+    return new Promise(resolve => {
+        setTimeout(resolve, ms)
+    })
+}
+
+let fetchJobStatus = async (jobId) => {
+    // fetch job status from bonsai api
+    const response = await fetch(`${apiURL}/job/status/${jobId}`)
+    return {status: response.status, data: await response.json()}
+}
+
+function validateJobStatus(result) {
+    // check job status
+    // returns true if run is invalid
+    let isValid = false
+    const jobInfo = result.data
+    if ( jobInfo.status === "finished" ) {
+        // if job has finished report result
+        isValid = true
+    } else if ( jobInfo.status === "failed" ) {
+        // if job failed raise error
+        throw new Error(`Job failed: ${jobInfo.result}`)
+        isValid = true
+    } 
+    console.log(`Job status: ${jobInfo.status}; is valid ${isValid}`)
+    return !isValid
+}
+
+// get result from job info object
+const resultParser = (result) => result.data.result
+
+// functions for hide/ showing spinner
+const showSpinner = (element) => {
```

Comment:
To clarify - this is not a criticism on this code. It is what it is when using pure JS.

Reply:
I agree. That is one of the drawbacks with vanilla JS. While I like react we decided to use flask and vanilla js for now since more colleuges knows how to work with it.